### DISCUSSION TITLE
Map base interface of source property to a class

### DISCUSSION THREAD

**Discussion by sunghwan2789** [riok/mapperly#319] [Q&A] [answered]
I want to map source property with a base interface to a class.
Specifically a fragment interface that defines all of the properties.

Given source:
```cs
public class Dto {
    public ISubDto Sub { get; }
}
public interface ISubDto : ISubFragment {
}
public interface ISubFragment {
    public int Id { get; }
}

public class Model {
    public SubModel Sub { get; set; }
}
public class SubModel {
    public int Id { get; set; }
}
```

Can I configure Mapperly to map `Dto.Sub` as `ISubFragment`?

There is a workaround via proxy methods, but it would be nice if Mapperly could detect this case:

```cs
[Mapper]
public static partial class DtoMapper {
    public static partial Model ToModel(this Dto source);

    public static partial SubModel ToSubModel(this ISubFragment source);

    private static SubModel MapToSubModel(ISubDto source) => ToSubModel(source);
}
```

**Comment by latonz** [comment, marked as answer]
This should just work, just remove the `ToSubModel` and `MapToSubModel` methods from the mapper.

**Reply by sunghwan2789** [reply]
I need `ToSubModel` and `MapToSubModel` to remove `RMG012` errors.

I am using editorconfig with [strict mappings](https://mapperly.riok.app/docs/configuration/analyzer-diagnostics#strict-mappings), and adding a new property to `SubModel` errors `RMG012`:
```diff
 public class SubModel {
     public int Id { get; set; }
+    public int UnmappedProperty { get; set; }
 }
```

```cs
[Mapper] // RMG012: Property UnmappedProperty on source type ISubDto was not found
public static partial class DtoMapper {
    public static partial Model ToModel(this Dto source);

    // public static partial SubModel ToSubModel(this ISubFragment source);

    // private static SubModel MapToSubModel(ISubDto source) => ToSubModel(source);
}
```

Can I resolve `RMG012` without removing them?

**Reply by latonz** [reply]
With strict mappings you should ignore unmapped properties using `MapperIgnoreTarget` or `MapperIgnoreSource`.
Eg. add this method:
```c#
[MapperIgnoreTarget(nameof(SubMode.UnmappedProperty))]
private static partial SubModel ToSubModel(ISubDto source);
```
